_sections/120-entities.md

```diff
@@ -163,6 +171,15 @@ When a consumer of this specification applies an entity `update`, it:
 - MUST treat a `__version` value other than a positive integer, including a missing value, as 0
 - MUST increment the `__version` of its local entity representation by 1 when an update is successfully applied
 
+*Added in spec version 2024.1.0*
+
+The `entity` element for an offline-capable update additionally:
+- MUST have a `branchId` attribute which is populated with a [RFC 4122 version 4 UUID](https://www.rfc-editor.org/rfc/rfc4122) generated by clients with an offline Entity representation.
+  - Clients that do have an offline Entity representation MUST keep the same `branchId` for a given Entity for an entire offline update sequence. They MUST generate a new `branchId` for a new update sequence after receiving a server update.
+  - When a `branchId` attribute is present in a submission, servers SHOULD attempt to process submissions with that same attribute in the correct order
```

Comment:
My only question is, after now having more experience with this feature in practice do you think that this needs to be a `SHOULD` rather than a `MUST`? Like as a ODK spec is there a plausible scenario where this aspect is not complied with.

Reply:
Thanks for the review!

I put this in mostly for other server implementations -- it's not a critical part of system function and they could choose that it's not all that important because it's not likely to be common. It feels helpful to call out things like this which could be cut from an implementation while still maintaining the spirit of the spec.

If you feel strongly, I'm happy to change it. It doesn't affect any of the core ODK tools.